### DISCUSSION TITLE
Struggling with the return value of a function

### DISCUSSION THREAD

**Discussion by m90att** [ExperienceLovelace/ha-floorplan#261] [Q&A] [answered]
Hi,

I'm creating my first function and I'm getting the function to work and return a value however the value returned contains a / (slash) instead of a '.'

I'm guessing that I need to escape it somewhere but I'm not familiar with doing that so I need some guidance please!

```
      functions: |
        >
        return {

          media_player_button_action: (entity) => {
            let text = entity;
            let switch_entity = text.replace("media_player.", "script.ents_");
            console.log(switch_entity);
            return switch_entity;
          },

        };
```

called by 

`service: ${functions.media_player_button_action(entity.entity_id)}  `

where the parameter passed is 'media_player.study' for example, and the console log reads

`script.ents_study`

which is the desired outcome.  

But, I get the following error in HA...

Failed to call service script/ents_study. Service not found.

I'm sure it's easy if you know but it's been puzzling me for long enough that I have to give in and ask for help!!

Thanks, Matt

**Comment by OzGav** [comment, marked as answer]
No that is what HA shows as I recall. Are you sure you have a script called `script.ents_study`

**Reply by m90att** [reply]
That's strange.  Yes, I've quadruple-checked!

![image](https://user-images.githubusercontent.com/56609577/201040476-9db80e1c-c4b4-4eb6-8a81-0f3d00672dc5.png)

**Reply by m90att** [reply]
I've worked out the issue.  I renamed the script (almost immediately after creating it a while ago) but it seems the underlying identity of it doesn't change from the original id given.  When I updated it, I changed the Entity ID as the screenshot above.  Digging into it a bit HA was still calling it script.switch_or_play_study which was the original name.

I created a new one from scratch and it works.  I'm really impressed with this functionality @exetico - great work, I've now used the new execute service and now functions.  Here it is below, have a little game of 'Spot the Difference'!

![image](https://user-images.githubusercontent.com/56609577/201043697-6ea2f734-cd86-414d-a1f3-93cc144c0bf9.png)

**Reply by OzGav** [reply]
Glad you got it sorted. it would be great if you could mark it as answered

**Reply by exetico** [reply]
@m90att That's great to hear! Thanks :)
